src/main/java/seedu/address/model/task/DateTimeDue.java

```diff
@@ -0,0 +1,78 @@
+package seedu.address.model.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import seedu.address.commons.util.DateTimeUtil;
+
+/**
+ * Represents the date-time when a Task is due.
+ * Guarantees: immutable; is an optional attribute of task;
+ * is valid as declared in {@link #isValidDateTimeDue(String)}
+ */
+public class DateTimeDue {
+
+    public static final String MESSAGE_CONSTRAINTS = "Due dates should adhere to one of the "
+            + "following formats:\n"
+            + "1. yyyyMMdd (e.g. 20201230)\n"
```

Comment:
i'm not sure why it cannot be yyyyMd, is there a difference?

Reply:
Interestingly, `yyyyMd` does not work.

Consider the date "12 Jan 2020", for instance. With `yyyyMd`, it would be "2020112". But this string is ambiguous. Are we referring to "2 Nov 2020" or "12 Jan 2020"?

If we want to use `yyyyMd`, we would need a delimiter (e.g. `yyyy-M-d`). I decided against using the hyphen because it doesn't read as well when it is a date range.